### PR TITLE
[Finishes #146028835] Allowed users to duplicate services

### PR DESCRIPTION
**Pull request by chalettu**
@miq-bot add_label enhancement
@miq-bot add_label fine/no
@miq-bot add_label catalogs / dialogs
@miq-bot add_label  ux/review

See https://github.com/ManageIQ/manageiq-ui-service/issues/937

### CODE REVIEW

**Comment by chalettu** [comment]
editdialog
Here is a screenshot of what duplicating an existing service from our Order explorer looks like.  From a UX perspective it wasn't much of a change.

**Comment by chriskacerguis** [comment]
@serenamarie125 @Loicavenel for your review!

**Comment by Loicavenel** [comment]
@chalettu Video is confusing, you are taking are clicking on Order but ending up in the Service, I don't see how you select a service in the order?

**Comment by chalettu** [comment]
new_reorder
Updated Screenshot for reordering of a service.  @Loicavenel we tried to push the button further right but can't due to limitations of it being an expanded content row of the Order above it. There is a certain amount of padding in the patternfly component that we can't override.

**Comment by chalettu** [comment]
@serenamarie125 here is the updated screenshot . For your review.

File: client/app/states/catalogs/details/details.state.js [hunk 1]
```diff
@@ -15,6 +15,19 @@ function getStates () {
...
+        serviceTemplate: resolveServiceTemplate,
+        serviceRequest: resolveServiceRequest
+      }
+    },
+    'catalogs.duplicate': {
+      url: '/duplicate/:serviceRequestId',
+      templateUrl,
+      controller: Controller,
+      controllerAs: 'vm',
+      title: __('Service Template Details'),
```
Comment (AllenBW) [inline client/app/states/catalogs/details/details.state.js]: Thinking this title should probably change 🎋

**Comment by serenamarie125** [comment]
Cool @chalettu ... any chance that Reorder button can be right aligned on the list view ?

**Comment by chalettu** [comment]
@serenamarie125 , that button is pushed as far right as it can get.  It is an issue with the patternfly listview and having listviews in the expanded contents.  We looked into it and its a matter of all the padding that gets added into the patternfly list view component

**Comment by chalettu** [comment]
@serenamarie125 , please let me know if my answer from yesterday makes sense and if we can proceed.

**Comment by chriskacerguis** [comment]
@serenamarie125 does @chalettu make sense, or should I setup a meeting to chat about that?

/cc @Loicavenel

**Comment by chriskacerguis** [comment]
@serenamarie125 does @chalettu make sense, or should I setup a meeting to chat about that?

/cc @Loicavenel

**Comment by chriskacerguis** [comment]
@chalettu please rebase and fix conflicts.

**Comment by chalettu** [comment]
@chriskacerguis , with greenkeeper updating as frequently as it is , I was going to hold off on rebasing frequently until we get the green light from UX on this PR.    It makes a lot of sense to only do one maybe two rounds of rebasing just to save time from having to do it daily until we get approval.

**Comment by serenamarie125** [comment]
@chalettu understood on the placement of the button.  We can handle that later.

Couple questions - is the reorder flow embedded in a dialog ?  I can't tell.  Also can't see the buttons to actually add to the shopping cart, so curious how this works?

**Comment by chalettu** [comment]
@serenamarie125 , The reorder flow is demonstrated in that screenshot.  We reused the same screen that was built for when you attempt to order a new service from the service catalog screen.  The difference is that we are filling the dialogs with the values that existed for the service you are attempting to duplicate.  The add to shopping cart button is at the top right of the dialog which hasn't changed from the existing screen we currently have for ordering a service.  It can be seen briefly in that screenshot demo in the top right part of the screen before we scroll down in the demo.   The flow for this process matches the exact flow for how a new service is ordered.

**Comment by serenamarie125** [comment]
That's great about using the same screen, etc and filling the default values.  Exactly what we want!
@chalettu Sorry, it's tough for me to differentiate if that is a dialog or embedded in the page.  If in a dialog, the action buttons should be on the bottom right and consistent with the rest of the SUI dialogs.

**Comment by Loicavenel** [comment]
@chalettu one question on my side, does it use the previous valued entered (what we want) or the reset to default? thanks

**Comment by chalettu** [comment]
@serenamarie125 the screenshot is using our dialog component on a page.  In the context of your last comment, are you intending to say the difference between being a modal vs being part of the page?   This particular screenshot is part of a larger page and is not a modal.  That being said, are we fine with the button placement ?

**Comment by chalettu** [comment]
@Loicavenel This uses the previous value entered for the service you are trying to duplicate

**Comment by Loicavenel** [comment]
Champagne!!

**Comment by serenamarie125** [comment]
@chalettu If it's not a modal, what does the Navigation look like? Can you increase the size of the image so we can see that, as well as show where the user is left after they click on "Add to Shopping Cart"

**Comment by chalettu** [comment]
add_to_cart
@serenamarie125 here is a bigger screen shot of the actual add to cart process.

**Comment by chalettu** [comment]
@serenamarie125 , did you have any more thoughts on this PR after reviewing the last screenshot and comments I submitted?

**Comment by serenamarie125** [comment]
@chalettu just to make sure i understand, the user goes to "My Orders", finds a request to Reorder, and then is moved to the Service Catalog area?  I think that's a bit funky, and that's why I was suggesting that it be put into a modal.  @Loicavenel what are your thoughts here, are you ok with the current flow?

**Comment by Loicavenel** [comment]
@serenamarie125 moving to the service Catalog does not make any sense. the way it was implemented in the second animation is perfect for me.  You go to My oder, you find an order, you expend it, then you select the service you want to re-order... click re-oder, show the dialog right there with overlay..

**Comment by serenamarie125** [comment]
@miq-bot remove_label ux/review

**Comment by serenamarie125** [comment]
@miq-bot add_label ux/approved

**Comment by chriskacerguis** [comment]
@AllenBW had previous approved this (but, it was dismissed due to a UX change), so going to merge.

**Comment by chriskacerguis** [comment]
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1502307

**Pull request merged**